qiskit/circuit/quantumcircuit.py

```diff
@@ -5089,6 +5089,15 @@ def _qasm2_define_custom_operation(operation, existing_gate_names, gates_to_defi
         )
     else:
         parameters_qasm = ""
+
+    # Gate definitions with 0 qubits or with any classical bits are not allowed.
+    if operation.num_qubits == 0 or operation.num_clbits != 0:
+        raise CircuitError(
+            "OpenQASM 2 does not allow gate definitions with no qubits or with any "
+            f"classical bit: '{operation.name}' has {operation.num_qubits} qubits "
+            f"and {operation.num_clbits} clbits"
+        )
```

Comment:
The error should be `QASM2ExportError`, but otherwise this is fine.

Not sure the comment is needed, since the error message says the exact same thing ;)

Reply:
Good catch. Thanks!